content/guides/dapps/journal.md

```diff
@@ -0,0 +1,564 @@
+  );
+}
+```
+
+Lastly, update the UI in `journal-ui.tsx` to take in a user input values for the
+`message` of when updating a journal entry:
+
+```typescript
+function JournalCard({ account }: { account: PublicKey }) {
+  const {
+    accountQuery,
+    updateEntry,
+    deleteEntry
+  } = useJournalProgramAccount({ account });
+  const { publicKey } = useWallet();
+  const [message, setMessage] = useState('');
+  const title = accountQuery.data?.title;
+
+  const isFormValid = message.trim() !== '';
+
+  const handleSubmit = () => {
+    if (publicKey && isFormValid && title) {
+      updateEntry.mutateAsync({ title, message, owner: publicKey });
+    }
+  };
+
+  if (!publicKey){
+    return <p>Connect your wallet</p>
+  }
+
+  return accountQuery.isLoading ? (
+    <span className="loading loading-spinner loading-lg"></span>
+  ) : (
+    <div className="card card-bordered border-base-300 border-4 text-neutral-content">
+      <div className="card-body items-center text-center">
+        <div className="space-y-6">
+          <h2
+            className="card-title justify-center text-3xl cursor-pointer"
+            onClick={() => accountQuery.refetch()}
+          >
+            {accountQuery.data?.title}
+          </h2>
+          <p>
+          {accountQuery.data?.message}
+          </p>
+          <div className="card-actions justify-around">
+            <textarea
+              placeholder="Update message here"
+              value={message}
+              onChange={(e) => setMessage(e.target.value)}
+              className="textarea textarea-bordered w-full max-w-xs"
+            />
+            <button
+              className="btn btn-xs lg:btn-md btn-primary"
+              onClick={handleSubmit}
+              disabled={updateEntry.isPending || !isFormValid}
+            >
+              Update Journal Entry {updateEntry.isPending && '...'}
+            </button>
+          </div>
+          <div className="text-center space-y-4">
+            <p>
+              <ExplorerLink
+                path={`account/${account}`}
+                label={ellipsify(account.toString())}
+              />
+            </p>
+            <button
+              className="btn btn-xs btn-secondary btn-outline"
+              onClick={() => {
+                if (
+                  !window.confirm(
+                    'Are you sure you want to close this account?'
+                  )
+                ) {
+                  return;
+                }
+                const title = accountQuery.data?.title;
+                if (title) {
+                  return deleteEntry.mutateAsync(title);
+                }
+              }}
+              disabled={deleteEntry.isPending}
+            >
+              Close
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+```
+
+## Resources
+
+- Journal dApp:
+  [solana-journal-eight.vercel.app](solana-journal-eight.vercel.app)
+- Program example code:
```

Comment:
There is no repo link in here

Reply:
Currently its not in program examples, making a PR with the code now

Reply:
FYI PR is here: https://github.com/solana-developers/dApp-examples/pull/1

Reply:
actually ignore that, the PR is now here: [https://github.com/solana-foundation/CRUD-dApp/pulls](https://github.com/solana-foundation/CRUD-dApp/pulls)

Reply:
added URL